### PR TITLE
Update Duplicate Detection to include Name and Phone

### LINKED ISSUE #59: Add Detect Duplicates
### Files update details:
{To be updated}

### PR DESCRIPTION
**Pull request by DennieDan**
Closes #59 
Note for @TY1Fan 
Currently, the test cases in `PersonTest` is like the following.
As when you change name to case insensitive, you need to change the `Name#equals` also. 
Please help to change the test from `assertFalse` to `assertTrue`. Thanks bro
```java
    @Test
    public void isSamePerson() {
        // same object -> returns true
        assertTrue(ALICE.isSamePerson(ALICE));

        // null -> returns false
        assertFalse(ALICE.isSamePerson(null));

        // same name and phone, all other attributes different -> returns true
        Person editedAlice = new PersonBuilder(ALICE).withEmail(VALID_EMAIL_BOB)
                                                     .withAddress(VALID_ADDRESS_BOB).withSchedule(VALID_SCHEDULE_BOB)
                                                     .withSubject(VALID_SUBJECT_BOB).withRate(VALID_RATE_BOB).build();
        assertTrue(ALICE.isSamePerson(editedAlice));

        // same name, all other attributes different -> returns false
        editedAlice = new PersonBuilder(ALICE).withPhone(VALID_PHONE_BOB).withEmail(VALID_EMAIL_BOB)
                .withAddress(VALID_ADDRESS_BOB).withSchedule(VALID_SCHEDULE_BOB)
                .withSubject(VALID_SUBJECT_BOB).withRate(VALID_RATE_BOB).build();
        assertFalse(ALICE.isSamePerson(editedAlice));

        // different name, all other attributes same -> returns false
        editedAlice = new PersonBuilder(ALICE).withName(VALID_NAME_BOB).build();
        assertFalse(ALICE.isSamePerson(editedAlice));

        // name differs in case, all other attributes same -> returns false
        // Note for Yi Fan: when you update the case-insensitive, this should return true
        Person editedBob = new PersonBuilder(BOB).withName(VALID_NAME_BOB.toLowerCase()).build();
        assertFalse(BOB.isSamePerson(editedBob));

        // name has trailing spaces, all other attributes same -> returns false
        String nameWithTrailingSpaces = VALID_NAME_BOB + " ";
        editedBob = new PersonBuilder(BOB).withName(nameWithTrailingSpaces).build();
        assertFalse(BOB.isSamePerson(editedBob));
    }

```

### CODE REVIEW

File: src/test/java/seedu/address/model/person/PersonTest.java [hunk 1]
```diff
@@ -35,16 +35,24 @@ public void isSamePerson() {
         // null -> returns false
         assertFalse(ALICE.isSamePerson(null));
 
-        // same name, all other attributes different -> returns true
-        Person editedAlice = new PersonBuilder(ALICE).withPhone(VALID_PHONE_BOB).withEmail(VALID_EMAIL_BOB)
-                .withAddress(VALID_ADDRESS_BOB).withSchedule(VALID_SCHEDULE_BOB).build();
+        // same name and phone, all other attributes different -> returns true
+        Person editedAlice = new PersonBuilder(ALICE).withEmail(VALID_EMAIL_BOB)
+                                                     .withAddress(VALID_ADDRESS_BOB).withSchedule(VALID_SCHEDULE_BOB)
+                                                     .withSubject(VALID_SUBJECT_BOB).withRate(VALID_RATE_BOB).build();
```
Comment (TY1Fan) [inline src/test/java/seedu/address/model/person/PersonTest.java:41]: I think there is too much indentation for these 2 lines.

**Review by TY1Fan — CHANGES_REQUESTED** [review]
I think the indentation at line 40 and 41 can be lesser. Else LGTM. Also thanks for the note for PersonTest.

File: src/test/java/seedu/address/model/person/PersonTest.java (hunk 1 above)
Comment (DennieDan) [reply src/test/java/seedu/address/model/person/PersonTest.java]: Okay, fixed

**Review by TY1Fan — APPROVED** [review]
LGTM!

**Pull request merged**